test/stdlib/TestUserInfo.swift

```diff
@@ -20,15 +20,29 @@ import StdlibUnittest
 class TestUserInfoSuper : NSObject { }
 #endif
 
-struct SubStruct {
+struct SubStruct: Equatable {
     var i: Int
     var str: String
+
```

Comment:
Yeah I determined the test wanted to check for value-equality, and not object-equality. It's literally the only way this assertion makes sense: https://github.com/apple/swift/blob/master/test/stdlib/TestUserInfo.swift#L105

Unless the test *really* intended to rely on the pointer-stability-even-from-different-bridgings fact, in which case it's just testing for something we no longer guarantee.